src/zkevm_specs/evm/instruction.py

```diff
@@ -817,3 +818,59 @@ def memory_copier_gas_cost(
         gas_cost = word_size * GAS_COST_COPY + memory_expansion_gas_cost
         self.range_check(gas_cost, N_BYTES_GAS)
         return gas_cost
+
+    def word_shift_right(self, a: RLC, shift: RLC) -> RLC:
+        assert len(a.le_bytes) == 32, "Expected argument a to contain 32 bytes"
+        assert len(shift.le_bytes) == 32, "Expected argument shift to contain 32 bytes"
+
+        # If shift is greater than 255, returns 0.
+        shift_lo, shift_hi = shift.le_bytes[:1], shift.le_bytes[1:]
+        shift_valid = self.is_zero(self.sum(shift_hi))
+        shift_bits = self.select(shift_valid, self.bytes_to_fq(shift_lo), FQ(0))
+        a = self.select(shift_valid, a, RLC(0))
+
+        # Return original value directly if shift is zero.
+        return self.select(
+            self.is_zero(shift_bits),
+            a,
+            self.word_shift_right_with_integer(U256(a.int_value), shift_bits.n),
+        )
+
+    def word_shift_right_with_integer(self, a: U256, shift: int) -> RLC:
```

Comment:
we shouldn't switch to `U256` and `int`, but instead stay in `FQ` when writing constraints.